toolium/utils/dataset.py

```diff
@@ -0,0 +1,124 @@
+        return param.replace("[NOW]", str(datetime.datetime.utcnow().strftime(date_format)))
+    elif "[TODAY]" in param:
+        return param.replace("[TODAY]", str(datetime.datetime.utcnow().strftime(date_day_format)))
+    elif "[TRUE]" in param:
+        return True
+    elif "[FALSE]" in param:
+        return False
+    elif "[NULL]" in param:
+        return None
+    elif type_mapping_match_group and type_mapping_match_group.group(1) \
+            in ["LIST", "DICT", "INT", "FLOAT", "STR", "UNICODE"]:
+        if type_mapping_match_group.group(1) == "STR":
+            return type_mapping_match_group.group(2)
+        elif type_mapping_match_group.group(1) == "UNICODE":
+            return unicode(type_mapping_match_group.group(2))
+        else:
+            exec(u"exec_param = {type}({value})".format(type=type_mapping_match_group.group(1).lower(),
+                                                        value=type_mapping_match_group.group(2)))
+            return locals()['exec_param']
+    elif date_matcher and len(date_matcher.groups()) == 3:
+        configuration = dict([(date_matcher.group(3).lower(), int(date_matcher.group(2).replace(" ", "")))])
+        now = (date_matcher.group(1) == 'NOW')
+        reference_date = datetime.datetime.utcnow() if now else datetime.datetime.utcnow().date()
+        replace_value = reference_date + datetime.timedelta(**configuration)
+        return replace_value.strftime(date_format) if now else replace_value.strftime(date_day_format)
+    else:
+        new_param = generate_fixed_length_param(param)
+
+    return new_param
+
+
+def generate_fixed_length_param(param):
+    """
+    Generate a fixed length param if the elements matches the expression
+    [<type>_WITH_LENGTH_<length>]. E.g.: [STRING_WITH_LENGTH_15]
+    :param param: Lettuce param
+    :return param with the desired length
+    """
+    try:
+        if "_WITH_LENGTH_" in param:
+            if "_ARRAY_WITH_LENGTH_" in param:
+                seeds = {'STRING': 'a', 'INTEGER': 1}
+                seed, length = param[1:-1].split("_ARRAY_WITH_LENGTH_")
+                param = list(seeds[seed] for x in xrange(int(length)))
+            elif "JSON_WITH_LENGTH_" in param:
+                length = int(param[1:-1].split("JSON_WITH_LENGTH_")[1])
+                param = dict((str(x), str(x)) for x in xrange(length))
+            else:
+                seeds = {'STRING': 'a', 'INTEGER': "1"}
+                # The chain to be generated can be just a part of param
+                start = param.find("[")
+                end = param.find("]")
+                seed, length = param[start + 1:end].split("_WITH_LENGTH_")
+                generated_part = seeds[seed] * int(length)
+                placeholder = "[" + seed + "_WITH_LENGTH_" + length + "]"
+                param = param.replace(placeholder, generated_part)
+                if seed == "INTEGER":
+                    param = int(param)
+    finally:
```

Comment:
May be it is better to remove try/finally to view exceptions when param is wrong defined